setup.py

```diff
@@ -1,51 +1,96 @@
-    
-    return parsed_requirements
-
-
-def get_dependency_links(filename):
+# 1. all packages should be listed here with their version requirements if any
+packages: List[str] = [
+    "farm==0.8.0",
+    "torch",  # torch package version already mentioned in farm
+    "fastapi",
+    "uvicorn",
+    "gunicorn",
+    "pandas",
+    "sklearn",
+    "psycopg2-binary",
+    "elasticsearch>=7.7,<=7.10",
+    "elastic-apm",
+    "tox",
+    "coverage",
+    "langdetect",  # for PDF conversions
+    "sentence-transformers",
+    "python-multipart",
+    "python-docx",
+    "sqlalchemy>=1.4.2",
+    "sqlalchemy_utils",
+    "faiss-cpu>=1.6.3",
+    "faiss-gpu",
+    "tika",
+    "uvloop==0.14",
+    "httptools",
+    "nltk",
+    "more_itertools",
+    "networkx",
+    "pymilvus",
+    "SPARQLWrapper",
+    "mmh3",
+    "weaviate-client",
+    "mypy",
+    "pytest",
+    "sentence-transformers",
+    "selenium",
+    "webdriver-manager",
+    "beautifulsoup4",
+    "markdown",
+    "streamlit",
+]
+# this is a lookup table with items like:
+# tokenizers: "tokenizers==0.9.4"
+# packaging: "packaging"
+dependencies: Dict = {b: a for a, b in (re.findall(r"^(([^!=<>]+)(?:[!=<>].*)?$)", x)[0] for x in packages)}
+
+
+def dependencies_list(*package_list):
     """
-     Parse a requirements pip file looking for the --find-links directive.
-    Args:
-        filename:  pip requirements requirements
-
-    Returns:
-        list of find-links's url
+    Return package with version from list
     """
-    with open(filename) as file:
-        parsed_requirements = file.read().splitlines()
-    dependency_links = list()
-    for line in parsed_requirements:
-        line = line.strip()
-        if line.startswith('--find-links'):
-            dependency_links.append(line.split('=')[1])
-    return dependency_links
-
-
-dependency_links = get_dependency_links('requirements.txt')
-parsed_requirements = parse_requirements('requirements.txt')
+    return [dependencies[package] for package in package_list]
+
+
+install_requires = [
+    dependencies["farm"],
+    dependencies["torch"],
+    dependencies["fastapi"],
+    dependencies["uvicorn"],
+    dependencies["gunicorn"],
+    dependencies["pandas"],
+    dependencies["sklearn"],
+    dependencies["psycopg2-binary"] + "; sys_platform != 'win32' and sys_platform != 'cygwin'",
+    dependencies["mmh3"],
+    dependencies["sqlalchemy"],
+    dependencies["sqlalchemy_utils"],
+    dependencies["networkx"],
+    dependencies["uvloop"] + "; sys_platform != 'win32' and sys_platform != 'cygwin'",
+]
+
+extras_require = {}
+extras_require["elasticsearch"] = dependencies_list("elasticsearch")
+extras_require["faiss-cpu"] = dependencies_list("faiss-cpu")
+extras_require["faiss-gpu"] = dependencies_list("faiss-gpu")
+extras_require["weaviate"] = dependencies_list("weaviate-client")
+extras_require["ui"] = dependencies_list("streamlit")
+extras_require["crawling"] = dependencies_list("selenium", "webdriver-manager")
```

Comment:
I believe beautifulsoup is also just used for crawling, but please verify

Reply:
beautifulsoup is used in markdown reader as well, but it is under try/catch
https://github.com/deepset-ai/haystack/blob/master/haystack/file_converter/markdown.py